src/python/pants/notes/2.17.x.md

```diff
@@ -1,5 +1,54 @@
 # 2.17.x Release Series
 
+## What's New
+
+### Overall
+
+The [`scie-pants` launcher binary](https://github.com/pantsbuild/scie-pants) is now the recommended way to run Pants.
+Use of a `./pants` shell script in a repository is deprecated. Among other benefits, the `scie-pants` launcher is able
+to automatically download and maintain a Python distribution for running Pants. It also allows you to run `pants`
+from any directory within a repository and not just the build root. See the [Pants  installation
+instructions](https://www.pantsbuild.org/docs/installation) for additional information on how to migrate to `scie-pants.`
+
+### Backends
+
+#### Python
+
+Python dependencies are now extracted via an engine intrinsic for increased performance.
```

Comment:
Is there a link we can point readers at to learn more about the improvements?

Reply:
cc @thejcannon

Reply:
I don't think we have one that isn't the PR itself

Reply:
Added link to the PR.